src/contracts/middleware/Middleware.sol

```diff
@@ -322,20 +316,54 @@ contract Middleware is
     function performUpkeep(
         bytes calldata performData
     ) external override checkAccess {
+        if (performData.length == 0) {
+            revert Middleware__NoPerformData();
+        }
+
         StorageMiddleware storage $ = _getMiddlewareStorage();
         address gateway = $.gateway;
         if (gateway == address(0)) {
             revert Middleware__GatewayNotSet();
         }
 
-        uint48 currentTimestamp = Time.timestamp();
-        if ((currentTimestamp - $.lastTimestamp) > $.interval) {
-            $.lastTimestamp = currentTimestamp;
+        uint48 epoch = getCurrentEpoch();
+        StorageMiddlewareCache storage cache = _getMiddlewareStorageCache();
+
+        address[] memory activeOperators = _activeOperators();
+        uint256 activeOperatorsLength = activeOperators.length;
+        uint256 cacheIndex = cache.epochToCacheIndex[epoch];
+        uint256 pendingOperatorsToCache = activeOperatorsLength - cacheIndex;
+
+        if (pendingOperatorsToCache > 0) {
```

Comment:
I would like us to have a test for the following case:
1. Make sure all the operators are already cached.
2. Repeat the last call with ValidatorsData.

Since all operators are cached already, it will go the second branch of the if, trying to decode as an array of bytes32. I want to make sure that fails and we decode it as garbage by mistake.